### ISSUE TITLE
unexpected '$' expecting '`', a new line followed by the next instruction, or the variable name

### ISSUE THREAD

**Issue by mloskot** [hadolint/hadolint#876]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

I'm not certain where is the problem but `docker` is happy to build images based on Windows ServerCore from my `Dockerfile`,
so I'd expect the linter accepts the `EXPOSE` port range specification in it (`Dockerfile` below):

```
${APP_PORT}-${APP_PORT_MAX}/tcp
```

### Actual behavior

```
bash-5.1# hadolint --no-fail --no-color test/Dockerfile 
test/Dockerfile:23:13 unexpected '$' expecting '`', a new line followed by the next instruction, or the variable name
```

I understand the line `${APP_PORT}-${APP_PORT_MAX}/tcp` in this bit

```
EXPOSE `
${APP_PORT}/tcp `
${APP_PORT}-${APP_PORT_MAX}/tcp
```

can be a bit unusual specification of port range, but Docker is fine about.

Simplifying it this way does not help:

```
EXPOSE ${APP_PORT}/tcp ${APP_PORT}-${APP_PORT_MAX}/tcp
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
# hadolint --version
Haskell Dockerfile Linter 2.10.0
```

**Dockerfile** (if relevant)

```Dockerfile
# escape=`
ARG BASE_IMAGE
ARG BASE_IMAGE_TAG

FROM ${BASE_IMAGE}:${BASE_IMAGE_TAG}
ARG ENTRYPOINT
ARG APP_VERSION
ENV APP_VERSION=${APP_VERSION}
WORKDIR /
COPY `
${ENTRYPOINT} `
scripts\app.ps1 `
./

ENV `
APP_PATH=C:\app.ps1 `
APP_PORT=4330 `
APP_PORT_MAX=4340

EXPOSE `
${APP_PORT}/tcp `
${APP_PORT}-${APP_PORT_MAX}/tcp

VOLUME ["D:", "Z:"]

ENTRYPOINT ["powershell.exe", "-NoProfile", "-ExecutionPolicy", "Bypass", "C:\\appps1;"]
```

**Comment by m-ildefons** [comment]
Hi Mateusz,
thanks for the detailed bug report. Judging by [the spec alone](https://docs.docker.com/engine/reference/builder/#expose) port ranges aren't a thing at all, so please forgive that this has slipped past. It seems that although we do handle port ranges already, the case where the port range is given with variables isn't handled by the parser.
I suggest that as a workaround until this is fixed, you give the ports explicitly in the `EXPOSE` line.

**Comment by mloskot** [comment]
@m-ildefons 
> Judging by [the spec alone](https://docs.docker.com/engine/reference/builder/#expose) port ranges aren't a thing at all,

OTOH, documentation on `docker run --expose`  says "Expose a port or a range of ports" and given discussions I've read on numerous GitHub issues and SO Q&A's like [this](https://stackoverflow.com/a/28721643/151641), I assumed that the `EXPOSE` documentation is simply out of date:
- https://github.com/docker/docker.github.io/issues/7802
- https://github.com/docker/cli/issues/1562

**Comment by mloskot** [comment]
Thanks @m-ildefons & @lorenzo
